src/sql/src/plan/statement/ddl.rs

```diff
@@ -2461,9 +2459,19 @@ fn kafka_sink_builder(
     sink_from: GlobalId,
 ) -> Result<StorageSinkConnection<ReferencedConnection>, PlanError> {
     let item = scx.get_item_by_resolved_name(&connection)?;
-    // Get Kafka connection
-    let mut connection = match item.connection()? {
-        Connection::Kafka(connection) => connection.clone(),
+    // Get Kafka connection + progress topic
+    //
+    // TODO: In ALTER CONNECTION, the progress topic must be immutable
```

Comment:
Is there an existing issue for this TODO?

Reply:
No, it's just something that exists in an unmerged PR: https://github.com/MaterializeInc/materialize/pull/21109/commits/a3e8fd40ac1a066ce95307727fc2f77d78c10b5c

Will make sure to remove the TODO.